capa/ida/plugin/form.py

```diff
@@ -725,23 +735,12 @@ def slot_progress_feature_extraction(text):
                     update_wait_box(f"{text} ({self.process_count} of {self.process_total})")
                     self.process_count += 1
 
-                update_wait_box("initializing feature extractor")
-
-                try:
-                    extractor = CapaExplorerFeatureExtractor()
-                    extractor.indicator.progress.connect(slot_progress_feature_extraction)
-                except Exception as e:
-                    logger.error("Failed to initialize feature extractor (error: %s).", e, exc_info=True)
-                    return False
-
-                if ida_kernwin.user_cancelled():
-                    logger.info("User cancelled analysis.")
-                    return False
```

Comment:
I see what you're trying to do here by re-using `self.extractor` for both the program analysis view and rule generator view. But let's keep this code to keep both extractors separate as setting the slot in line `738` (right) also affects the rule generator UX.